src/Zipper.elm

```diff
@@ -0,0 +1,81 @@
+module Zipper exposing
+    ( Zipper
+    , getCurrent
+    , makeZipper
+    , moveBackwards
+    , moveBackwardsTo
+    , moveBackwardsWhile
+    , moveForward
+    , moveForwardTo
+    , moveForwardWhile
+    )
+
+import Debug
+import List exposing (append, head, isEmpty, tail)
+import List.Extra as ListE
+import Maybe exposing (withDefault)
+
+
+type alias Zipper a =
+    { previous : List a
+    , current : a
+    , next : List a
+    }
+
+
+makeZipper : List a -> a -> List a -> Maybe (Zipper a)
+makeZipper previous current next =
+    Just { previous = previous, current = current, next = next }
+
+
+moveForward : Zipper a -> Maybe (Zipper a)
+moveForward zipper =
+    Just <|
+        if isEmpty zipper.previous then
+            --TODO: figure out whether it should actually be returning "Nothing"
+            zipper
+
+        else
+            { previous = withDefault [] <| ListE.init zipper.previous
+            , current = withDefault zipper.current <| ListE.last zipper.previous
+            , next = zipper.current :: zipper.next
+            }
+
+
+moveBackwards : Zipper a -> Maybe (Zipper a)
+moveBackwards zipper =
+    Just <|
+        if isEmpty zipper.next then
+            --TODO: figure out whether it should actually be returning "Nothing"
+            zipper
+
+        else
+            { previous = append zipper.previous [ zipper.current ]
+            , current = withDefault zipper.current <| head zipper.next
+            , next = withDefault [] <| tail zipper.next
+            }
```

Comment:
This is identical to `moveForward`, and we don't like duplicates.
Can you figure out a way to avoid this issue?

Hint: What's the difference between
```elm
moveForward { previous = [], current = 1, next = [2] }
```
and
```elm
moveBackwards { previous = [2], current = 1, next = [] }
```
?

Reply:
The `previous` and `next` are switched, and if they had more arguments they will also be reversed.
Should I create a `flipZipp` function and make the `moveBackwards` be just a call for the `moveForwards` with the `flipZipp`?

Reply:
Or do the same without an additional function?